testing/enclient/enclient.go

```diff
@@ -0,0 +1,149 @@
+package enclient
+
+import (
+	"bytes"
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"log"
+	"math/big"
+	"net/http"
+	"time"
+
+	"github.com/google/exposure-notifications-server/internal/base64util"
+	"github.com/google/exposure-notifications-server/internal/database"
+)
+
+const (
+	dkLen               = 16
+	maxTransmissionRisk = 8
+	maxIntervals        = 144
+)
+
+type Interval int32
+
+// Posts requests to the specified url.
+// This methods attempts to serialize data argument as a json.
+func PostRequest(url string, data interface{}) (*http.Response, error) {
+	request := bytes.NewBuffer(JsonRequest(data))
+	r, err := http.NewRequest("POST", url, request)
+	if err != nil {
+		return nil, err
+	}
+	r.Header.Set("Content-Type", "application/json")
+	client := &http.Client{}
+	resp, err := client.Do(r)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return resp, fmt.Errorf("post request failed with status: %v", resp.StatusCode)
+	}
+
+	return resp, nil
+}
+
+// Serializes the given argument to json.
+func JsonRequest(data interface{}) []byte {
+	jsonData, err := json.Marshal(data)
+	if err != nil {
+		log.Fatalf("unable to marshal json payload")
```

Comment:
It's _probably_ fine since this is in `testing/`, but you'd usually want to return this error back to its caller, at which point you might as well move the `json.Marshal` call into the body.

Reply:
Ack. Leaving it like this to simplify code of the tests.